contract-examples/test/fee_manager.ts

```diff
@@ -9,257 +9,89 @@ import {
+
 // TODO: These tests keep state to the next state. It means that some tests cases assumes some preconditions
 // set by previous test cases. We should make these tests stateless.
 describe("ExampleFeeManager", function () {
   this.timeout("30s")
 
-  let owner: SignerWithAddress
-  let contract: Contract
-  let manager: SignerWithAddress
-  let nonEnabled: SignerWithAddress
-  let managerPrecompile: Contract
-  let ownerPrecompile: Contract
-  before(async function () {
-    owner = await ethers.getSigner(adminAddress);
-    const contractF: ContractFactory = await ethers.getContractFactory("ExampleFeeManager", { signer: owner })
-    contract = await contractF.deploy()
-    await contract.deployed()
-    const contractAddress: string = contract.address
-    console.log(`Contract deployed to: ${contractAddress}`)
-
-    managerPrecompile = await ethers.getContractAt("IFeeManager", FEE_MANAGER, manager);
-    ownerPrecompile = await ethers.getContractAt("IFeeManager", FEE_MANAGER, owner);
+  beforeEach("setup DS-Test contract", async function () {
+    const signer = await ethers.getSigner(ADMIN_ADDRESS);
+    const feeManagerPromise = ethers.getContractAt("IFeeManager", FEE_MANAGER, signer)
 
-    const signers: SignerWithAddress[] = await ethers.getSigners()
-    manager = signers.slice(-1)[0]
-    nonEnabled = signers.slice(-2)[0]
-
-    let tx = await ownerPrecompile.setEnabled(manager.address);
-    await tx.wait()
-
-    tx = await owner.sendTransaction({
-      to: manager.address,
-      value: ethers.utils.parseEther("1")
-    })
-    await tx.wait()
-
-    tx = await owner.sendTransaction({
-      to: nonEnabled.address,
-      value: ethers.utils.parseEther("1")
-    })
-    await tx.wait()
-  });
-
-  it("should add contract deployer as owner", async function () {
-    const contractOwnerAddr: string = await contract.owner()
-    expect(owner.address).to.be.equal(contractOwnerAddr)
-  });
-
-  it("contract should not be able to change fee without enabled", async function () {
-    let contractRole = await managerPrecompile.readAllowList(contract.address);
-    expect(contractRole).to.be.equal(ROLES.NONE)
-    try {
-      let tx = await contract.enableWAGMIFees()
-      await tx.wait()
-    }
-    catch (err) {
-      return
-    }
-    expect.fail("should have errored")
-  })
-
-  it("contract should be added to manager list", async function () {
-    let adminRole = await ownerPrecompile.readAllowList(adminAddress);
-    expect(adminRole).to.be.equal(ROLES.ADMIN)
-    let contractRole = await ownerPrecompile.readAllowList(contract.address);
-    expect(contractRole).to.be.equal(ROLES.NONE)
-
-    let enableTx = await ownerPrecompile.setEnabled(contract.address);
-    await enableTx.wait()
-    contractRole = await ownerPrecompile.readAllowList(contract.address);
-    expect(contractRole).to.be.equal(ROLES.ENABLED)
-  });
-
-  it("admin should be able to change fees through contract", async function () {
```

Comment:
I deleted this test as it is redundant. The same functionality is covered in `admin should be able to enable change fees` .

(TODO fix the wording in the new test)